jx_sqlite/expressions.py

```diff
@@ -249,11 +249,12 @@ def to_sql(self, schema, not_null=False, boolean=False):
     for t in "bsnj":
         if lhs[t] and rhs[t]:
             acc.append("(" + lhs[t] + ") = (" + rhs[t] + ")")
+            type = t
```

Comment:
this will not work when there is more than one type.  I suggest you append the null comparison to the equality test in the line above.

Or you may want to write `return Not("not", Eq("eq", [self.lhs, self.rhs])).to_sql(schema, not_null, boolean)`